infra/base-images/base-builder/compile

```diff
@@ -63,8 +63,24 @@ if [ $FUZZING_ENGINE = "none" ]; then
   export COVERAGE_FLAGS=
 fi
 
-export CFLAGS="$CFLAGS $SANITIZER_FLAGS $COVERAGE_FLAGS"
-export CXXFLAGS="$CFLAGS $CXXFLAGS_EXTRA"
+if [[ $FUZZING_LANGUAGE == "rust" ]]; then
```

Comment:
I'm confused why if-else is necessary here. Why can't these rustflags be defined in addition to the C/C++ ones so we don't need to parse the yaml files in helper.py?

Reply:
Reason is https://github.com/google/oss-fuzz/commit/bc45406906f5ffddf2b82663e7ecdef8ad679db9#diff-7a7598e90ce6e156a92aad21f6f27990L24, also i tried putting it in comments, as per project owners, it breaks stuff.

Reply:
I think it's a bit of a weird workflow for rust now that the project.yaml must be filled it out in order for the build to work.

Reply:
Max added it to presubmit long time back when he needed it for code coverage stuff. language is a required attribute and added in all projects. Having language there is important i feel, it makes us track what project types we have etc.
https://github.com/google/oss-fuzz/blob/master/infra/presubmit.py#L127
My breakages was trying to fix that language support to be properly available in all places.

Reply:
What about Suricata which is a project with both C and Rust ?

Reply:
I checked yesterday, we can actually get rid of FUZZING_LANGUAGE and still keep sanitizer flags without breaking rust projects (this is contrary to what developers had in their build.sh). I am unsure what will happen with code coverage support coming soon. What about keeping rust targets in a seperate project.

Reply:
Suricata project has fuzz targets written in C that will run C and Rust code.
I am not sure if MSAN can work with this

It seems it works with ASAN (I had caught a `panic` in Rust with a buggy fuzz target)